python/shotgun_fields/image_widget.py

```diff
@@ -258,7 +297,7 @@ def _on_worker_failure(self, uid, msg):
         """
         On failure just display an error and set the toolkit to the error string.
         """
-        if uid == self._task_uid:
+        if str(uid) == self._task_uid:
```

Comment:
why is this needed? is the `request_thumbnail` method return value broken?
Please add a comment to explain.